base/reinterpretarray.jl

```diff
@@ -391,8 +389,6 @@ check_ptr_indexable(a::Array, sz) = sizeof(eltype(a)) !== sz
 check_ptr_indexable(a::Memory, sz) = true
 check_ptr_indexable(a::AbstractArray, sz) = false
 
-@propagate_inbounds getindex(a::ReinterpretArray) = a[firstindex(a)]
```

Comment:
It seems simpler to replace L394 with
```julia
@propagate_inbounds getindex(a::ReshapedReinterpretArray{T,0}) where {T} = a[firstindex(a)]
```
?

Reply:
thank you for the review!

would `only` be acceptable here rather than `a[firstindex(a)]` ? I know it is not necessary because a 0 dimensional array has only a single index, but in my opinion that would make it extra-clear that this is a special case